### ISSUE TITLE
incorrectly counts the time of sleep. zetime

### ISSUE THREAD

**Issue by ARMrus** [Freeyourgadget/Gadgetbridge#1701]
#### Before reporting a bug, please confirm the following:
- [x] I have read the [wiki](https://github.com/Freeyourgadget/Gadgetbridge/wiki), and I didn't find a solution to my problem / an answer to my question.
- [x] I have searched the [issues](https://github.com/Freeyourgadget/Gadgetbridge/issues), and I didn't find a solution to my problem / an answer to my question.
- [x] If you upload an image or other content, please make sure you have read and understood the [github policies and terms of services](https://help.github.com/articles/github-terms-of-service/#1-responsibility-for-user-generated-content)

### I got Gadgetbridge from:
* [x] F-Droid
* [ ] I built it myself from source code (specify tag / commit)

If you got it from Google Play, please note [that version](https://github.com/TaaviE/Gadgetbridge) is unofficial and not supported here; it's also often quite outdated. Please switch to one of the above versions if you can.

#### Your issue is:
On [skrine](https://armrus.org/~gq5yQ) in sight, slept with 3:14 until 11:00 time. The app Gadgetbridge counted only 30 minutes.

#### Your wearable device is:
ZeTime
ver: A2.0 (r3.0T3.3H0.5) build57

#### Your android version is:
Android 9 LineagoOS 16.0

#### Your Gadgetbridge version is:
0.37.1

**Comment by Lightwars** [comment]
Yes we only count the light and deep sleep. The ZeTime App had this definition of sleep data:

- SLEEP_TYPE_AWAKE = 2;
- SLEEP_TYPE_BEGIN = 16;
- SLEEP_TYPE_DEEP = 0;
- SLEEP_TYPE_END = 17;
- SLEEP_TYPE_EXIT_SLEEP = 4;
- SLEEP_TYPE_LIGHT = 1;
- SLEEP_TYPE_READY_SLEEP = 3;

Gadgetbridge only have the activity kinds of light and deep sleep, so the others are of type unkown. Because I do not know what to do with them. Maybe some are counting also for sleep... or the value returned from the watch have changed with a firmware update, I do not know at the moment.

How about doing a comparising of Gadgetbridge and one of the MyKronoz apps?

1. record your sleep
2. before syncing check the box which allows to let the data on the watch in the ZeTime preferences
3. sync with Gadgetbridge
4. now connect with one MyKronoz app
5. compare the sleep graphs and look for the deep sleep and light sleep minutes

Please report back here.

**Comment by ARMrus** [comment]
Hopefully user ZeTime will do that. Watch ZeTime is not available to me now :(

**Comment by ARMrus** [comment]
The clock came back :)
When the application MyKronoz data from the clock is erased, so there is [only a screen of the sleep schedule.](https://armrus.org/~6jH4M)

**Comment by ARMrus** [comment]
How to get the log of communication hours with a smartphone via Bluetooth?

**Comment by Lightwars** [comment]
> How to get the log of communication hours with a smartphone via Bluetooth?

On Android you can activate to log bluetooth communication within the developers options. It is called "Enable BluetoothHCI Snoop Log" or something similar. Enable it, log the communication you are interested in. disable it and copy the log file to a pc. open it with wireshark and you can see, what is going on. 
Have a look [here](https://github.com/Freeyourgadget/Gadgetbridge/wiki/ZeTime-Protocol-Analysis), to see what is going on between your phone and the ZeTime.

**Comment by ARMrus** [comment]
Enabled Bluetooth HCI in android settings. The log file itself does not know where to look?

**Comment by Lightwars** [comment]
> Enabled Bluetooth HCI in android settings. The log file itself does not know where to look?

This depends on device vendor or android version. For me it is here: "/data/misc/bluetooth/logs/btsnoop_hci.log"

It is possible that this file "/etc/bluetooth/bt_stack.conf" says where to store the log file, for example there is a line like this: "BtSnoopFileName =/sdcard/btsnoop_hci.log".

Note: you need root to copy this log or access the location.

**Comment by ARMrus** [comment]
Yes, there is a large binary file. How to read it now? If decoder for this file?

**Comment by Lightwars** [comment]
> Yes, there is a large binary file. How to read it now? If decoder for this file?



> On Android you can activate to log bluetooth communication within the developers options. It is called "Enable BluetoothHCI Snoop Log" or something similar. Enable it, log the communication you are interested in. disable it and copy the log file to a pc. **open it with wireshark and you can see, what is going on.**
> Have a look [here](https://github.com/Freeyourgadget/Gadgetbridge/wiki/ZeTime-Protocol-Analysis), to see what is going on between your phone and the ZeTime.
